internal/sponge/item/model.go

```diff
@@ -32,3 +44,18 @@ func (item *Item) Validate() error {
 
 	return nil
 }
+
+// InferIDFromData infers an item_id from type and source id.
+func (item *Item) InferIDFromData() error {
+
+	// If a source id value is found, use it to compose the item_id for this item.
+	idValue, ok := item.Data[sourceIDField]
+	if !ok {
+		return fmt.Errorf("Cannot Infer ID: Unable to find source id field")
+	}
+
+	// Set the ID via the template.
+	item.ID = fmt.Sprintf(itemIDTemplate, item, idValue)
```

Comment:
This may have been unclear.  itemIDTemplate is a string containing the FORMAT_SPECIFIER. I moved the string itself into the call for readability. It's only used there so no need to duplicate strings.